### PR TITLE
Expander

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###
Moved Expander from XF to XCT.

### Bugs Fixed ###
- Related to issue #214 

### API Changes ###

Added: 
- Expander
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/expander

### Behavioral Changes ###
None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of main at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: XamarinCommunityToolkit/Views/Expander/ExpandDirectionExtensions.shared.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	static class ExpandDirectionExtensions
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/Expander/ExpandDirectionExtensions.shared.cs:4]: I made it internal for now

File: XamarinCommunityToolkit/Views/Expander/ExpandDirectionExtensions.shared.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	{
+		public static bool IsVertical(this ExpandDirection orientation)
+			=> orientation == ExpandDirection.Down
+				|| orientation == ExpandDirection.Up;
+
+		public static bool IsRegularOrder(this ExpandDirection orientation)
```
[the remark this answers is not in the document]
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkit/Views/Expander/ExpandDirectionExtensions.shared.cs:10]: Oh, actually it doesn't "horizontal/vertical"
It checks if orientation is "Left To Righ" or "Top to Bottom"

[the remark this answers is not in the document]
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkit/Views/Expander/ExpandDirectionExtensions.shared.cs:10]: That's why I called it "regular" order.

File: XamarinCommunityToolkitSample/Xamarin.CommunityToolkit.Sample.csproj [hunk 3]
```diff
@@ -19,6 +19,7 @@
       <AutoGen>True</AutoGen>
       <DependentUpon>AppResources.es.resx</DependentUpon>
     </Compile>
+    <Compile Remove="Resx\ExpanderPage.xaml.cs" />
```
[the remark this answers is not in the document]
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkitSample/Xamarin.CommunityToolkit.Sample.csproj]: done

**Comment by jfversluis** [comment]
Did you need to change anything for this to work? No, right?

I think there were some issues opened on the Forms repo, are there any enhancements in here already? Or should we open issues here and point to the Forms ones to fix them?

**Comment by AndreiMisiukevich** [comment]
> Did you need to change anything for this to work? No, right?
> 
> I think there were some issues opened on the Forms repo, are there any enhancements in here already? Or should we open issues here and point to the Forms ones to fix them?

This implementation contains "Direction" support.
Also, I renamed one enum "ExpanderState" to "ExpandState"

**Comment by AndreiMisiukevich** [comment]
> Did you need to change anything for this to work? No, right?
> 
> I think there were some issues opened on the Forms repo, are there any enhancements in here already? Or should we open issues here and point to the Forms ones to fix them?

most of the issues are related to CollectionView. I don't think we can fix them in expander

**Comment by jfversluis** [comment]
Ah OK, cool! I'll have a look at this tomorrow, but since this is copied 1:1 mostly I don't foresee any issues.

Maybe we could write a little transition guide for people who were already using this.

0. Install Toolkit
1. Remove experimental flag
2. Change namespace
3. If you're using `ExpanderState` rename to `ExpandState`

Anything I'm missing?

**Comment by AndreiMisiukevich** [comment]
> Ah OK, cool! I'll have a look at this tomorrow, but since this is copied 1:1 mostly I don't foresee any issues.
> 
> Maybe we could write a little transition guide for people who were already using this.
> 
> 1. Install Toolkit
> 2. Remove experimental flag
> 3. Change namespace
> 4. If you're using `ExpanderState` rename to `ExpandState`
> 
> Anything I'm missing?

No) Awesome guide 👍

**Review by jfversluis — APPROVED** [review]

**Comment by jfversluis** [comment]
@AlexHedley this should be an easy move from one Doc to another :) 

For someone finding this, wrote a little transition guide here if you come from Xamarin.Forms: https://github.com/xamarin/XamarinCommunityToolkit/wiki/Features-moved-from-Xamarin.Forms#expander

File: XamarinCommunityToolkit/Views/Expander/Expander.shared.cs [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+				(Abs(height - previousSize.Height) >= double.Epsilon && !Direction.IsVertical()))
+				ForceUpdateSize();
+
+			previousSize = new Size(width, height);
+		}
+
+		protected override void OnChildAdded(Element child)
+		{
+			base.OnChildAdded(child);
+			if (mainLayout == null && child is StackLayout layout) {
```
Comment (pictos) [inline XamarinCommunityToolkit/Views/Expander/Expander.shared.cs:243]:
```suggestion
			if (mainLayout == null && child is StackLayout layout)
			{
```

**Review by sthewissen — APPROVED** [review]

**Pull request merged**